### PR TITLE
  feat: Add generic CreateTask job for automated task creation

### LINKED ISSUE #1343: [Feature Request] Add corejob for task creation
Tasks within the core api allow for easy tracking an assignment of work that needs to be completed. There are many instances where a task should be created automatically, for example, based on a policy review due date, a task would be created and assigned round-robin to a user in the approver group of a policy. This might also be used to create things like "onboarding tasks". 

This issue is to add the basic job that could be used to create the task on behalf of the user. 

Job should be added to the `corejobs` pkg, see the [README](https://github.com/theopenlane/core/blob/main/pkg/corejobs/README.md) for help getting started. 

Job arguments should include:
* Type - this will allow for the code that is doing the insert to just specify "Policy Review", "Evidence Review", or "Onboarding" , etc and this job would define all the details of what would be needed for that job type. Additionally, we could have an "Generic" type, where the title, description, category (optional) and assignee could be passed. To get this started, lets start with the following types:
    * Generic - Provide title, description as required, category and assignee as optional 
    * Policy Review - Provide internal policy id, which will create a task that has:
        * Title: `<Policy Name> Review`
        * Description: Conduct the annual review of this internal policy to ensure it remains accurate, effective, and aligned with current business practices, legal requirements, and compliance frameworks. Review and update the content as needed, obtain necessary approvals, and document any changes or confirmations. 
        * Category: `Policy Review`
        * internalPolicyIDs: `[]string{<ulid of policy>}`
        * Assignee: Grab a random user from the policy `approver` group 
* Organization ID - all requests should include the organization id that this task will be created within (this will be the `OwnerID` on the request)
* Depending on the type, other details might be needed, e.g. Policy Review would require the `InternalPolicyID` to be sent. Onboarding would require the `OrganizationID`. 
* Delay (Optional) - Schedule the job to run in a future time/date

This job should use the openlane client to create the task. See other jobs in [corejobs](https://github.com/theopenlane/core/tree/main/pkg/corejobs) for examples of using the client within a job. 

Ensure the config for the worker has:
```
type TaskWorkerConfig struct {
	Enabled bool `koanf:"enabled" json:"enabled" jsonschema:"required description=whether the task worker is enabled"`

	OpenlaneAPIHost  string `koanf:"openlaneAPIHost" json:"openlaneAPIHost" jsonschema:"required description=the openlane api host"`
	OpenlaneAPIToken string `koanf:"openlaneAPIToken" json:"openlaneAPIToken" jsonschema:"required description=the openlane api token"`
}
```

Documentation: 
River: https://riverqueue.com/docs/scheduled-jobs
Openlane create task: https://docs.theopenlane.io/docs/api/graph-api/inputObjects#createtaskinput
Riverboat: repo that jobs are added to, this is for reference - no need to add this here as part of this task: https://github.com/theopenlane/riverboat

### PR DESCRIPTION
**Pull request by manab-pr**
Add a flexible, generic task creation job to the corejobs package that enables programmatic task creation from external schedulers and triggers.

This job follows a separation of concerns design where:
- External systems (schedulers, polling jobs, triggers) handle business logic like fetching policies, selecting assignees, and building task details
- This job simply creates tasks with the provided details

This approach makes it configurable and reusable across different use cases like policy reviews, onboarding workflows, evidence uploads, and any future scenarios without requiring code changes.

Key features:
- Generic task creation accepting all standard task fields
- Required: title, description, organization_id
- Optional: category, assignee_id, assigner_id, due_date, tags, internal_policy_ids, and more
- Scheduled job support via ScheduledAt field
- Comprehensive validation and error handling
- 11 test cases covering happy paths, validation, and edge cases

Closes #1343

### CODE REVIEW

File: pkg/corejobs/create_task.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+// GetOpenlaneConfig returns an OpenlaneConfig from the TaskWorkerConfig
+func (c *TaskWorkerConfig) GetOpenlaneConfig() *OpenlaneConfig {
+	return &OpenlaneConfig{
+		OpenlaneAPIHost:  c.OpenlaneAPIHost,
+		OpenlaneAPIToken: c.OpenlaneAPIToken,
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (golanglemonade) [reply pkg/corejobs/create_task.go]: this is a valid suggestion ^  but it should just be 

```
	// embed OpenlaneConfig to reuse validation and client creation logic
	OpenlaneConfig
```
without the ``koanf:",squash" json:",inline"``

File: pkg/corejobs/create_task.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	// Validate required fields
+	if err := w.validateArgs(job.Args); err != nil {
+		logger.Error().Err(err).Msg("invalid job arguments")
+		return err
+	}
+
+	// Initialize Openlane client if not already set
+	if w.olClient == nil {
+		olConfig := w.Config.GetOpenlaneConfig()
+		cl, err := olConfig.getOpenlaneClient()
```
[the remark this answers is not in the document]
Comment (golanglemonade) [reply pkg/corejobs/create_task.go]: ^ also valid

[the remark this answers is not in the document]
Comment (manab-pr) [reply pkg/corejobs/create_task.go]: done

File: pkg/corejobs/create_task.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (manab-pr) [reply pkg/corejobs/create_task.go]: done as suggested.
